### ISSUE TITLE
XPath returns a node instead of an attribute

### ISSUE THREAD

**Issue by dennwc** [bblfsh/sdk#340]
From the new Python client:

> ... There is a failing test that I've left commented out, `testFiltertoken`, PTAL, this xpath query:
>
>```python
>it = ctx.filter("//*[@token='else']/@token")
>```
>
> AFAIK it should return the `@token` property value (`else`) but it returns the full object containing it.

See https://github.com/bblfsh/client-python/pull/128#issuecomment-446199039

Related either to XPath projection in SDK, or XPath library itself.

**Comment by juanjux** [comment]
The problem is probably the `@` in the `@token` (and `@type`, `@pos`, etc...) since XPath uses the `@` symbol to retrieve attribute values.

**Comment by dennwc** [comment]
`_BBLFSH_` is too hard to use, so maybe we can pick something different?

The token specifically is mapped to the node's content. Since does might have both token and children, I also projected is as a separate child node (`<token>` or `<Token>`?).

`@type` is projected to a node element name, so we don't need to handle it that way.

`@pos` is projected to attributes as: `<pos-name>-start-offset`, etc.

And we don't have any other `@` fields defined.

**Comment by juanjux** [comment]
Yeah, `_BBLFSH` it was just an example, but the user would still use `@` and it would be translated internally (would still be ugly on the fixtures). The length is to avoid possible collisions with drivers; existing properties. But since we translate whatever field is the token in the native AST, we should be right using another "token" property for this specific case.

**Comment by creachadair** [comment]
Is this an issue that arises from queries generated inside the library (rather than directly by users), and if so can we work around it by adding quotes explicitly? I'm pretty sure XPath allows "@" in values, as long as they're escaped.

**Comment by juanjux** [comment]
Nope, no scaping of symbols for XPath 1.0 as far as I investigated. Double quotes doesn't work either. This would also affect, actually I found about this while upgrading the Python client's unittests.

The solution proposed by @dennwc of creating internally a 'token' property with the @token should be simple to implement.

**Comment by dennwc** [comment]
P.S. I believe it was already implemented and merged ;) We just haven't updated the docs.
